scale/scale_test.go

```diff
@@ -0,0 +1,113 @@
+package scale
+
+import (
+	"github.com/stretchr/testify/assert"
+	"testing"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/apis/extensions"
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/watch"
+)
+
+type MockDeployment struct {
```

Comment:
ok, now i see what you meant about the code duplication... this is maybe a bit more code than i would have duplicated, or at least you could put it in its own file so it doesn't clutter the real test file. if you really do want it in the same file, at least put it *after* the tests so the most important code in the file is at the top.